Sources/ConcurrencyHelpers/Event.swift

```diff
@@ -0,0 +1,93 @@
+import Atomics
+
+/**
+ * Synchronization primitive modelled after Event from WinAPI.
+ */
+public final class Event {
```

Comment:
Just asking, why don't do it as actor? Seems it will simplify the logic since it will be guarantee to be race free.

Reply:
Performance.
Want to able to check `isSignalled` as fast as possible.

Reply:
Well, don’t get me wrong, but seems if actor is not often occupied (which in this use case it will not be) it should take the same time. For me it looks like optimisation without clear benefit but with more complex logic.
Or maybe we should add the performance test to see difference and consider proper patter going forward in other places [](edited-with-Axolo)

Reply:
Well, performance numbers are always good to have.
But frankly I cannot imagine anything faster than reading a flag from memory.
And since this code is in general-purpose library I think it makes sense to trade a bit of complexity for performance.

Another reason I don't use actors -- I want methods be available in sync context.

Reply:
> Well, performance numbers are always good to have.

Yeah, so maybe will try to have some numbers and see? :slightly_smiling_face:
Regarding sync context I would agree, sometimes it’s useful [](edited-with-Axolo)